department-of-veterans-affairs/abd-vro
# PDF Generator: Part 3 - Hypertension

# PDF Generator: Part 3 - Hypertension 

## Description

**NOTE:** This is part of the work from #108 which is being broken down into more manageable PRs. Trying to keep PRs under 300 changes

This is the full list of PRs that make up the PDF Generator:
- Consumer - #126 
- Asthma Generation - #127
- Hypertension Generation - #128
- S3 Upload - #129



### What was the problem?

PDF Generator was missing Hypertension option

### How does this fix it?

By passing in the request template from #113 as a message to RabbitMQ, the consumer will generate a PDF for Hypertension(7701).

## How to test this PR

- `./gradlew clean`
- `./gradlew build check docker`
- `./gradlew :app:dockerComposeDown :app:dcPrune :app:dockerComposeUp`
- Once all containers are up, navigate to service-python/examples
- Run `python generate_pdf.py`. You can edit the `code` field to be either `6602` or `7701`
- View docker logs to see that the consumer received the message and displays the bytes of the generated PDF. **NOTE**: The PDF is currently not being saved to any location in this PR. It is just being displayed in the logs
-  Tests can be run using Pytest by entering the docker container or creating your own virtualenv with the `requirements.txt` installed and then running `python -m pytest`
